platform-validation/scripts/compatibility.sh

```diff
@@ -357,11 +356,53 @@ perform_cleanup(){
     #TODO : Cleanup docker images
 }
 
+check_kernel_file(){
+if [ ! -e "$CONFIG" ]; then
+	wrap_warning "warning: $CONFIG does not exist, searching other paths for kernel config ..."
+	for tryConfig in $POSSIBLE_CONFIGS; do
+		if [ -e "$tryConfig" ]; then
+			CONFIG="$tryConfig"
+			break
+		fi
+	done
+	if [ ! -e "$CONFIG" ]; then
+		wrap_warning "error: cannot find kernel config"
+		wrap_warning "  try running this script again, specifying the kernel config:"
+		wrap_warning "    CONFIG=/path/to/kernel/.config $0 or $0 /path/to/kernel/.config"
+		EXIT_CODE=1
+	fi
+fi
+}
+
 
 # ------------------------------------------------------------------------------
 # Check whether the Target Device can be used to set capability. EdgeHub   Runtime component sets CAP_NET_BIND which is Required for Azure IoT Edge Operation.
 # ------------------------------------------------------------------------------
 
+check_flag() {
+	if is_set "$1"; then
+		wrap_pass "CONFIG_$1"
+    else
+		wrap_fail "CONFIG_$1"
+	fi
+}
+
+check_kernel_flags(){
+    EXIT_CODE=0
+    check_kernel_file
+    if [ $EXIT_CODE != 0 ]; then 
+        wrap_fail "check_kernel_flags"
+        return
+    fi
+    EXIT_CODE=0
+    wrap_debug "Reading Kernel Config from $CONFIG"
+    for flag in "$@"; do
```

Comment:
$@ corresponds to all the command line arguments right? The plan was to pass the first argument as "config path" like ./compatibility.sh "/path1 /path2 /path3" or ./compatibility.sh "/path1"  "/path2"  "/path3"

Reply:
No, $@ refers to the arguments passed to the functions themselves